pkg/cmd/node/node_metrics.go

```diff
@@ -0,0 +1,112 @@
+package node
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-cli/pkg/cli"
+	"github.com/dcos/dcos-core-cli/pkg/metrics"
+	"github.com/dcos/dcos-core-cli/pkg/pluginutil"
+	"github.com/spf13/cobra"
+)
+
+func newCmdNodeMetrics(ctx api.Context) *cobra.Command {
+	var jsonOutput bool
+	cmd := &cobra.Command{
+		Use:       "metrics",
+		Short:     "Print details or summary of the metrics of a specified agent",
+		ValidArgs: []string{"details", "summary"},
+		Args:      cobra.ExactArgs(2),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if args[0] != "details" && args[0] != "summary" {
+				return fmt.Errorf("the first argument must be 'details' or 'metrics' e.g. dcos node metrics details <mesos-id> [--json]")
```

Comment:
`'details' or 'metrics'` >  `'details' or 'summary'`

However, shall we rather create 2 subcommands (`details` and `summary`)? It removes the need for manual validation and separates the code (looking at the body of the command it looks like it's 2 totally different code paths).